app/controllers/texts_controller.rb

```diff
@@ -1,6 +1,10 @@
 class TextsController < ApplicationController
+  PER_PAGE = 10
   def index
     genre_list = ["Basic", "Git", "HTML&CSS", "Ruby", "Ruby on Rails"]
-    @texts = Text.where(genre: genre_list)
+
+    @search = Text.ransack(genre: genre_list)
+    @texts = @search.result.page(params[:page]).per(PER_PAGE)
```

Comment:
これですと，検索フォームの内容が無視されてしまいます。先にwhereで絞り込みましょう。

```suggestion
    @search = Text.where(genre: genre_list).ransack(params[:q])
    @texts = @search.result.page(params[:page]).per(PER_PAGE)
```